source/docs/example-projects/halloween-treasure-hunt.rst

```diff
@@ -0,0 +1,251 @@
+* :doc:`Section 2.2 - The Basics of Denizen Commands </docs/basics-of-scripting/basics-of-denizen-commands>`
+* :doc:`Section 2.7 - Your First World Script </docs/basics-of-scripting/your-first-world-script>`
+* foreach loops here
+
+This will help you make a better use of this example project, and follow its contents successfully.
+
+.. note::
+    It is also recommended to have the `meta documentation <https://one.denizenscript.com/denizen/logs>`_ open in 
+    another tab while scripting this project, as we'll need to check the syntax of events, commands and tags very 
+    often.
+
+First Steps
+===========
+
+Now let's jump (or slowly crawl) into the scripting process. In our case, we want to make a **Halloween Treasure 
+Hunt** event in our *Hub world*. We plan on manually placing some hidden pumpkins ourselves, and reward players for 
+finding (and left clicking) them.
+
+First of all, we'll need to create a *new script file* (for example ``Halloween_Treasure_Hunt.yml``) and build a world 
+script container which, as we know, will listen to events happening within our server. We just have to give it a name, 
+like ``Halloween_Treasure_Hunt``, and set the ``type:`` key to ``world``. In addition to this, we'll also enable debug 
+by setting the ``debug:`` key to ``true``. This will make our script print in the console everything it does and help 
+us solve errors. It is not needed, so we can just disable it later on. We'll now add an ``events:`` subkey, which will 
+in the end hold the executable code we're going to write.
+
+Our file with the script container looks like this:
+
+.. code-block:: dscript
+    :name: figure_1_1
+    :linenos:
+    :emphasize-lines: 1-4
+
+    Halloween_Treasure_Hunt:
+    type: world
+    debug: true
+    events:
+
+.. rst-class:: figurecaption
+
+**Figure 1.1** Our starting world script container
+
+It's worth noting that Denizen scripts follow **YAML**'s indentation rules, with either *2 or 4-space tabs*. Many text 
+editors can be used for writing these scripts, but we personally prefer **Notepad++**. You just have to make sure the 
+tabs are replaced with actual spaces. There's a settings option in that automatically does that for us, so no problem.
+
+Now we need to find an event that fits our case. Looking through the *event list* documentation, ``on player clicks 
+block`` looks like our best bet. We just have to adjust it for our specific use case by adding *optional* arguments. 
+According to the event syntax, it accepts a click type, a material and an area, which will just be ``left``, 
+``pumpkin`` and ``in Hub`` respectively.
+
+.. note::
+    This tutorial assumes our *Hub world* is in fact called ``Hub``.
+
+We'll just make sure it works for now, so we add it under the events key and end the line with a ``:``. Inside this 
+event we can place our first command. A good option for testing purposes is just to show a word in *chat*. The command 
```

Comment:
A better way to word this is probably "A good way to see if something works is to print something to either your chat or to look at the debug in your console." The user needs to be aware of both equally viable debugging options.